### ISSUE TITLE
Ruby 2.7 support

### ISSUE THREAD

**Issue by ekohl** [ploubser/JSON-Grep#46]
When running jgrep under Ruby 2.7, the following warning can be seen:
```
/home/ekohl/.gem/ruby/gems/jgrep-1.5.2/lib/parser/parser.rb:84: warning: deprecated Object#=~ is called on FalseClass; it always returns nil
```

There may be more

**Comment by bastelfreak** [comment]
Hi people. I see the same warning a lot, which makes shell reading relly hard. besides that I also get:

```
.vendor/ruby/2.7.0/gems/jgrep-1.5.2/lib/jgrep.rb:180: warning: deprecated Object#=~ is called on Integer; it always returns nil
```

**Comment by ripienaar** [comment]
@bastelfreak can you see if changing line 180 to this works for you?

```
if !tmp.nil? && value.is_a?(String) && value =~ /^\/.*\/$/
```

**Comment by ripienaar** [comment]
@ekohl I think for yours this should fix:

```
if !n_token.nil? && n_token.is_a?(String) && !n_token =~ /|and|or|not|\(/
```

or maybe 

```
if !n_token.nil? && !n_token.to_s =~ /|and|or|not|\(/
```

We might have to ask @ploubser nicely to look at the ones in parser though

**Comment by ploubser** [comment]
I'll take a look at this as soon as I'm free.

**Comment by ploubser** [comment]
I raised https://github.com/ploubser/JSON-Grep/pull/47 which fixes the deprecation warnings that the tests threw up under 2.7.0-preview2. 

@ekohl @bastelfreak would you be able to confirm that this fixes all the deprecation warnings that you saw? If it does I'll push a new release.

**Comment by ploubser** [comment]
Cool, this has been merged and I pushed a new 1.5.3 gem.

**Comment by ripienaar** [comment]
Thank you @ploubser

**Comment by ekohl** [comment]
Thanks!

**Comment by GabrielNagy** [comment]
Hello,

Unfortunately the Ruby 2.7 fixes did not get into the 1.5.3 release. We encountered similar warnings when testing the puppet_agent module with Ruby 2.7.

```
/home/gabi/.rvm/gems/ruby-2.7.1/gems/jgrep-1.5.3/lib/jgrep.rb:180: warning: deprecated Object#=~ is called on Float; it always returns nil
/home/gabi/.rvm/gems/ruby-2.7.1/gems/jgrep-1.5.3/lib/jgrep.rb:180: warning: deprecated Object#=~ is called on Integer; it always returns nil
```

To confirm this I manually unpacked the gem from https://rubygems.org/gems/jgrep/versions/1.5.3. Looks like the tag was made before the Ruby 2.7 PR.

**Comment by ploubser** [comment]
Hey @GabrielNagy. Big oof. I created and pushed a 1.5.4 that includes the changes. My bad.
